Update prompt builder documentation to agent builder

MuyangAmigo opened the pull request:
- Update Prompt Builder to Agent Builder
- Include new documentation for Tool (MCP Servers) use in Agent Builder
- Update screenshots and GIFs to match the latest UIs.

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+---
+ContentId: bd3d7555-3d84-4500-ae95-6dcd39641af0
+DateApproved:
```
```suggestion
DateApproved: 04/22/2025
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+---
+# Build agents and prompts in AI Toolkit
+
+   > [!NOTE]
```
```suggestion
> [!NOTE]
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+# Build agents and prompts in AI Toolkit
+
+   > [!NOTE]
+   > Agent Builder was previously known as Prompt Builder. The name has been changed to better reflect the feature's capabilities and its focus on building agents.
```
```suggestion
> Agent Builder was previously known as Prompt Builder. The name has been changed to better reflect the feature's capabilities and its focus on building agents.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+   > [!NOTE]
+   > Agent Builder was previously known as Prompt Builder. The name has been changed to better reflect the feature's capabilities and its focus on building agents.
+
+Agent builder in AI Toolkit streamlines the engineering workflow for building agents, including prompt engineering and integration with tool use such as MCP servers. It helps with common prompt engineering tasks:
```
```suggestion
Agent Builder in AI Toolkit streamlines the engineering workflow for building agents, including prompt engineering and integration with tools, such as MCP servers. It helps with common prompt engineering tasks:
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+- Break down complex tasks through prompt chaining and structured outputs
+- Provide easy access to code for seamless Large Language Model (LLM) integration via APIs
+
+As well as enhancing the intelligent app's capabilities with tool use, such as:
```
```suggestion
Agent Builder also enhances intelligent app's capabilities with tool use:
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+
+## Create, edit, and test prompts
+
+To access the agent builder, use either of these options:
```
```suggestion
To access Agent Builder, use either of these options:
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+- In the AI Toolkit view, select **Agent (Prompt) Builder**
+- Select **Try in Agent (Prompt) Builder** from a model card in the model catalog
+
+To test a prompt in the agent builder, follow these steps:
```
```suggestion
To test a prompt in Agent Builder, follow these steps:
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+   The *user prompt* is the input that you want to send to the model. The optional *system prompt* is used to provide instructions with relevant context to guide the model response.
+
+   > [!TIP]
+   > If you don't know how to input these prompts, you can describe your project idea in natural language and let the AI-powered feature generate prompts for you to experiment with.
```
```suggestion
   > You can describe your project idea by using natural language and let the AI-powered feature generate the prompts for you to experiment with.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+2. Select **MCP server project**
+3. Select a programming language from the dropdown list: **Python** or **TypeScript**
+4. Select a folder to create the new MCP server project in.
+5. Input a name for the MCP server project.
```
```suggestion
5. Enter a name for the MCP server project.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+You can also use the agent builder to test the MCP server. The agent builder will send the prompts to the MCP server and display the response.
+
+Follow these steps to test the MCP server:
+> Notes
```
```suggestion

> [!NOTE]
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+2. Use AI Toolkit Agent Builder to test the server with the following prompt:
+   1. System Prompt: You are a weather forecast professional that can tell weather information based on given location.
+   2. User Prompt: What is the weather in Shanghai?
+3. Server will be auto-connected to the Agent Builder.
```
```suggestion
3. The server will be automatically connected to Agent Builder.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+   1. System Prompt: You are a weather forecast professional that can tell weather information based on given location.
+   2. User Prompt: What is the weather in Shanghai?
+3. Server will be auto-connected to the Agent Builder.
+4. Click `Run` to test the server with the prompt.
```
```suggestion
4. Select `Run` to test the server with the prompt.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,133 @@
+
+![Use structured output](./images/promptbuilder/structured_output_2.gif)
+
+To test using structured output in the agent builder, follow these steps:
```
```suggestion
To test using structured output in Agent Builder, follow these steps:
```

MuyangAmigo commented:
> @MuyangAmigo Looking good. I've added some minor text edits.

@ntrogh Thanks for the review! I have updated the PR with your comments. We can hold off the merge until next Monday with stable release of AI Toolkit.

MuyangAmigo commented:
@ntrogh , @lszomoru I made slight updates on MCP server section per feedbacks we got for AITK. Please help take another look, thank you!

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -43,30 +43,55 @@ To test a prompt in Agent Builder, follow these steps:
-You can use the agent builder to connect to existing MCP servers or build a new MCP server from scaffold.
+You can use the agent builder to discover and configure featured MCP servers, connect to existing MCP servers or build a new MCP server from scaffold.
 
+### Configure MCP server from featured selections
```
```suggestion
### Configure a featured MCP server
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -43,30 +43,55 @@ To test a prompt in Agent Builder, follow these steps:
+AI Toolkit provides a list of featured MCP servers that you can use to connect to external APIs and services.
+
+To configure an MCP server from featured selections, follow these steps:
+1. In the **Tools** section, select **+ MCP Server** and select **+ Add server** in the quick pick.
```
```suggestion
1. In the **Tools** section, select **+ MCP Server**, and then select **+ Add server** in the Quick Pick.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -43,30 +43,55 @@ To test a prompt in Agent Builder, follow these steps:
+To configure an MCP server from featured selections, follow these steps:
+1. In the **Tools** section, select **+ MCP Server** and select **+ Add server** in the quick pick.
+2. Select **Use Featured MCP Servers** from the dropdown list.
+3. Select a MCP server that fit your needs.
```
```suggestion
3. Choose an MCP server that meets your needs.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -43,30 +43,55 @@ To test a prompt in Agent Builder, follow these steps:
+For servers that are ready to use, AI Toolkit will prompt you to select tools and add to your agent afterwards.
+
+For servers that are marked with **(Configuration Required)**:
+1. You will be prompted to enter the required configurations in `mcp.json` file.
```
```suggestion
1. You will be prompted to enter the required configurations in the `mcp.json` file.
```

ntrogh commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -43,30 +43,55 @@ To test a prompt in Agent Builder, follow these steps:
+   2. [GitHub](https://github.com/modelcontextprotocol/servers/tree/main/src/github#npx)
+   3. [Google Maps](https://github.com/modelcontextprotocol/servers/tree/main/src/google-maps#npx)
+   4. [Sqlite](https://github.com/modelcontextprotocol/servers/tree/main/src/sqlite#uv-1)
+3. Once completed the configuration, go back to the Agent Builder and select **+ MCP Server** to locate the server you added from the list.
```
```suggestion
3. After finishing the configuration, go back to Agent Builder and select **+ MCP Server** to locate the server you added from the list.
```

ntrogh reviewed (changes requested):
@MuyangAmigo a few minor text edits

kuojianlu commented on docs/intelligentapps/agentbuilder.md:
```diff
@@ -0,0 +1,158 @@
+3. Select a MCP server that fit your needs.
+![connect to MCP server](./images/promptbuilder/Featured_server.png)
+
+For servers that are ready to use, AI Toolkit will prompt you to select tools and add to your agent afterwards.
```
Featured server may need `npx` or `uv` dependency, AITK will validate if `npx` or `uv` is installed, if not, AITK will prompt user that `npx` or `uv` is not installed. Can we add this information here? And also provide the installation link for them:
`node`: https://nodejs.org/en/download
`npx`: `npm install -g npx`
`uv`: https://docs.astral.sh/uv/getting-started/installation/

When this information is added here, I can add a `Learn more` link in the `npx` or `uv` not installed prompt.

MuyangAmigo replied:
Updated

MuyangAmigo commented:
> @MuyangAmigo a few minor text edits

@ntrogh Thanks, I have updated the PR with your comments

ntrogh commented:
@MuyangAmigo let me know when the updates can be merged.

MuyangAmigo commented:
> @MuyangAmigo let me know when the updates can be merged.

@ntrogh The latest version of AITK has been released, we can merge docs now.

ntrogh commented:
@MuyangAmigo Docs updates are live
